ouroboros-network-framework/src/Ouroboros/Network/Subscription/Dns.hs

```diff
@@ -67,236 +53,82 @@ data Resolver m = Resolver {
     , lookupAAAA :: DNS.Domain -> m (Either DNS.DNSError [Socket.SockAddr])
     }
 
-
-dnsResolve :: forall m s.
+-- | Resolve a name. AAAA and A are done concurrently and waited on, then
+-- the results concatenated, with AAAA going before A.
+dnsResolve
+  :: forall m .
      ( MonadAsync m
-     , MonadSay   m
      , MonadSTM   m
-     , MonadTime  m
-     , MonadTimer m
+     , MonadCatch m
      , MonadThrow m
      )
-    => Tracer m DnsTrace
-    -> Resolver m
-    -> StrictTVar m s
-    -> BeforeConnect m s Socket.SockAddr
-    -> DnsSubscriptionTarget
-    -> m (SubscriptionTarget m Socket.SockAddr)
-dnsResolve tracer resolver peerStatesVar beforeConnect (DnsSubscriptionTarget domain _ _) = do
```

Comment:
I took out this happy eyeball thing for simplicity. Maybe we want it back? AFAICT though it's not worth it: how often are we doing a resolve-to-connect? Not very. It was intended to be used by lookup-intensive interactive applications like web browsers, no?